gsoft-inc/sg-orbit
# BREAKING: Add support for React 18

Issue: N/A

## Summary

This PR adds support for React 18, and drops support for React <=17.

## What I did

- Update Storybook to support React 18.
- Update `@types/react` to support React 18.
- Update `@testing-library` to support React 18.
- Removed `@testing-library/react-hooks`, since the version that supports React 18 has been merged into `@testing-library/react`
- Fix tests broken by React 18 and the new version of `@testing-library`, mostly caused by `act(() => ...)` not being awaited.
- Removed `@reach/useId`, to use React 18's `useId`.
- ~Add a polyfill for `useId`, for React versions lower than 18.~ Not anymore. We're okay with this being a **breaking change.**



## How to test

- Is this testable with Jest or Chromatic screenshots? ✅

- Does this need an update to the documentation? ❌

If your answer is yes to any of these, please make sure to include it in your PR.
